src/javascript/_common/os_detect.js

```diff
@@ -0,0 +1,23 @@
+const OSDetect = () => {
+    const Systems = {
+        macos  : ['Mac68K', 'MacIntel', 'MacPPC'],
+        linux  : ['HP-UX', 'Linux i686', 'Linux armv7l', ''],
+        ios    : ['iPhone', 'iPod', 'iPad'],
+        android: ['Android'],
+        windows: ['Win16', 'Win32', 'WinCE'],
+    };
+    if (typeof navigator !== 'undefined') {
+        return Object.keys(Systems)
+            .map(os => {
+                if (Systems[os].some(platform => navigator.platform === platform)) {
+                    return os;
+                }
+                return false;
```

Comment:
still suggest changing this to `.map(os => Systems[os].some(platform => navigator.platform === platform))`

Reply:
This only return boolean that which `index` of the `Object.keys()`'s array has the item. We have to search the array again to fetch `os` name